torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -227,6 +227,19 @@ def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
                 transformer_block = checkpoint_wrapper(
                     transformer_block, job_config.activation_checkpoint
                 )
+
+            if job_config.training.compile:
+                if (
+                    job_config.activation_checkpoint.mode == "selective"
+                    and job_config.activation_checkpoint.selective_ac_option == "op"
+                ):
+                    torch._dynamo.config._experimental_support_context_fn_in_torch_utils_checkpoint = (
+                        True
+                    )
+                logger.info(f"Compiling TransformerBlock {layer_id} with torch.compile")
```

Comment:
It's kinda hard, we have to put compile here bc we want to have compile apply to each TransformerBlock after AC, but before FSDP wrapping, to avoid the dynamic shape issue, otherwise 2D compile would still not work